### DISCUSSION TITLE
Additional Images attached to a Recipe

### DISCUSSION THREAD

**Discussion by nathanpalmer** [mealie-recipes/mealie#4134] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I'm testing out Mealie and so far it's been impressive. I'm converting over some of my notes of recipes and some of these are from my mom. There are multiple photos of her original 3x5 cards that were hand written. I want to keep these with the recipe if possible. 

### Please provide a concise description of the feature that would resolve your issue.

It makes sense to have a single highlight image for the recipe, I would like to have a way to add additional images for things like this.

### Please consider and list out some caveats or tradeoffs made in your design decision

The biggest tradeoff is additional complexity? But it could be a toggle on each recipe.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I like this idea, maybe allowing multiple images in a gallery/carousel

FWIW you can embed images in recipe instructions (and in the description, I think?) which might be sufficient, at least in the short term

**Reply by nathanpalmer** [reply]
I didn't realize you could add images there! Could be useful.

 It doesn't quite make sense in my case because I couldn't tie a single image to 1 set of instructions. If we were able to attach images to a note though, that could work. Because I could add a note that is "Original Notecards" and add both of them to it.

**Reply by YukonRob** [reply]
I am looking for the same exact thing, to add a photo of the original recipe as written.

**Comment by upendedshark** [comment]
Bumping this request. any chance it can be revisited? i use paprika and always save multiple images from recipes, or take multiple photos.

**Comment by Arne1303** [comment]
Is there any PR currently being worked on for this?
I really feel like the image handling for recipes could need a face lift to allow multiple images in a carousel (as suggested above by @michael-genson) and also the editing/cropping of images when being uploaded as a title image.

I am more than happy to start a PR for this if there is none currently being worked on!

**Comment by aortmannm** [comment]
Yes cropping and changing files a little bit which part is shown etc would be really great. Also a download button of the original file I searched last time would be great.

**Comment by aortmannm** [comment]
@Arne1303 did you get an answer if someone works on this?

**Reply by Arne1303** [reply]
So far I didnt get any reply! Im still open for an PR or you can go ahead with one if you want to.

**Reply by Sorilo** [reply]
I would also love a carousal feature to be implemented and ideally the option to download the selected image if possible. Did either of you start working on this feature/PR?

**Reply by Arne1303** [reply]
@Sorilo So far I didn't, write a message when you start that we don't have duplicated PRs!

**Comment by bferd** [comment]
Quick update: the first step of this shipped in v3.25.0 (https://github.com/mealie-recipes/mealie/pull/7933). Clicking the recipe hero image or an image asset now opens a full-screen lightbox with zoom and pan.

I've opened a draft PR for the first part of this: #8450. It adds prev/next navigation (arrows, keyboard, swipe, position chip) to the lightbox for the hero image and image assets.

Hero or asset image click: prev/next through the hero image followed by the image assets, in their existing order (no wrap-around)
Step image click: prev/next through every step image, in step order, with a "Step N" label
@Arne1303 if you've already started on anything carousel-related, let me know so we don't duplicate work. Cropping on upload would be a separate piece and isn't part of this.

Future plan: v3 would support video assets and step videos
